src/textual/widget.py

```diff
@@ -3103,8 +3120,16 @@ async def _on_compose(self) -> None:
         except Exception:
             self.app.panic(Traceback())
         else:
+            self._extend_compose(widgets)
             await self.mount(*widgets)
 
+    def _extend_compose(self, widgets: list[Widget]) -> None:
+        """Hook to extend composed widgets.
+
+        Args:
+            widgets: Widgets to be mounted.
+        """
+
```

Comment:
I don't get the overarching idea of adding `_extend_compose`. 😞

Reply:
It's to allow the Screen to add a Tooltip to the DOM, without requiring it to be explicitly added to `compose()`